lib/components/map/connected-stops-overlay.js

```diff
@@ -1,64 +1,62 @@
+import * as uiActions from '../../actions/ui'
 
 // connect to the redux store
 
-const mapStateToProps = (state, ownProps) => {
+const mapStateToProps = (state) => {
   const { mainPanelContent, viewedRoute } = state.otp.ui
-  const { routes } = state.otp.transitIndex
+  const { patternId, routeId } = viewedRoute || {}
+  const { patterns, v2 } = state.otp.transitIndex.routes?.[routeId] || {}
+  const visible = mainPanelContent !== uiActions.MainPanelContent.STOP_VIEWER
 
-  let { stops } = state.otp.overlay.transit
   let minZoom = 15
+  let stops = []
+
+  if (
+    mainPanelContent === uiActions.MainPanelContent.ROUTE_VIEWER &&
+    patterns
+  ) {
+    // Display stops for a pattern or all patterns for a route.
+    let viewedPattern = patternId
+
+    // If a flex route is being shown but the pattern viewer is not active, then the
+    // stops of the first pattern of the route should be shown
+    // This will ensure that the flex zone stops are shown.
+
+    // Preferably, the flex stops would be rendered in a separate layer.
+    // However, without changes to GraphQL, getting this data is very expensive
+    if (!viewedPattern && v2) {
+      viewedPattern = Object.keys(patterns)?.[0]
+    }
 
-  // There are a number of cases when stops within route objects should be shown
-  // although there are no stops that would otherwise be shown. In this case we need
-  // to override the stops index
-  if (routes) {
-    // All cases of stops being shown even when stops are otherwise hidden
-    // only apply when a route is being actively viewed
-    const route = routes?.[viewedRoute?.routeId]
-    if (route?.patterns) {
-      // If the pattern viewer is active, stops along that pattern should be shown
-      let viewedPattern = viewedRoute?.patternId
-      // If a flex route is being shown but the pattern viewer is not active, then the
-      // stops of the first pattern of the route should be shown
-      // This will ensure that the flex zone stops are shown.
-
-      // Preferably, the flex stops would be rendered in a separate layer.
-      // However, without changes to GraphQL, getting this data is very expensive
-      if (!viewedPattern && route.v2) {
-        viewedPattern = Object.keys(route?.patterns)?.[0]
+    // Discard duplicates from pattern stops.
+    stops = (patterns?.[viewedPattern]?.stops || []).reduce((prev, cur) => {
+      if (!prev.find((stop) => stop.id === cur.id)) {
+        prev.push(cur)
       }
-
-      // Override the stop index so that only relevant stops are shown
-      stops = route.patterns?.[viewedPattern]?.stops
-      // Discard duplicates
-      stops = (stops || []).reduce((prev, cur) => {
-        if (!prev.find((stop) => stop.id === cur.id)) {
-          prev.push(cur)
-        }
-        return prev
-      }, [])
-      // Override the minimum zoom so that the stops appear even if zoomed out
-      minZoom = 2
-    }
+      return prev
+    }, [])
+    minZoom = 2
```

Comment:
The comment about this got lost!

Reply:
Oops, I added the comment back in https://github.com/opentripplanner/otp-react-redux/pull/720/commits/6e6a9726464f4793b915a7c01c1bcd3c7c1d44cc.